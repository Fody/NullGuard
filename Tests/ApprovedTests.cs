#if (NET472)
using VerifyXunit;
using Fody;
using Xunit;
using System.Linq;
using System.Threading.Tasks;

[UsesVerify]
public class ApprovedTests
{
    [Fact]
    public Task ClassWithBadAttributes()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithBadAttributes"));
    }

    [Fact]
    public Task ClassWithPrivateMethod()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithPrivateMethod"));
    }

    [Fact]
    public Task ClassWithPrivateMethodNoAssert()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithPrivateMethod"));
    }

    [Fact]
    public Task GenericClass()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "GenericClass`1"));
    }

    [Fact]
    public Task GenericClassWithValueTypeConstraint()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "GenericClassWithValueTypeConstraints`1"));
    }

    [Fact]
    public Task GenericClassWithReferenceTypeConstraints()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "GenericClassWithReferenceTypeConstraints`1"));
    }

    [Fact]
    public Task Indexers()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "Indexers"));
    }

    [Fact]
    public Task InterfaceBadAttributes()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "InterfaceBadAttributes"));
    }

    [Fact]
    public Task SimpleClass()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleClass"));
    }

    [Fact]
    public Task SimpleClassNoAssert()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SimpleClass"));
    }

    [Fact]
    public Task SkipIXamlMetadataProvider()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "XamlMetadataProvider"));
    }

#if (DEBUG)
    [Fact]
    public Task SpecialClass_debug()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SpecialClass"));
    }
#else
    [Fact]
    public Task SpecialClass_release()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "SpecialClass"));
    }
#endif

    [Fact]
    public Task PublicNestedInsideNonPublic()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "NonPublicWithNested"));
    }

    [Fact]
    public Task UnsafeClass()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "UnsafeClass"));
    }

    [Fact]
    public Task ClassWithImplicitInterface()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithImplicitInterface"));
    }

    [Fact]
    public Task ClassWithExplicitInterface()
    {
        return Verifier.Verify(Ildasm.Decompile(AssemblyWeaver.AfterAssemblyPath, "ClassWithExplicitInterface"));
    }

    [Fact]
    public Task InfosList()
    {
        return Verifier.Verify(AssemblyWeaver.TestResult.Messages.Select(x=>x.Text));
    }

    [Fact]
    public Task WarnsList()
    {
        return Verifier.Verify(AssemblyWeaver.TestResult.Warnings.Select(x=>x.Text));
    }

    [Fact]
    public Task ErrorsList()
    {
        return Verifier.Verify(AssemblyWeaver.TestResult.Errors.Select(x=>x.Text));
    }
}

#endif