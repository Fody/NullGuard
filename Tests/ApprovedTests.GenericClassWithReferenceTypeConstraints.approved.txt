.class public auto ansi beforefieldinit GenericClassWithReferenceTypeConstraints`1<class T>
extends [mscorlib]System.Object
{
.field private !T '<NonNullProperty>k__BackingField'
.field private !T '<CanBeNullProperty>k__BackingField'
.method public hidebysig specialname instance !T
get_NonNullProperty() cil managed
{
.maxstack  2
IL_0000:  ldarg.0
IL_0001:  ldfld      !0 class GenericClassWithReferenceTypeConstraints`1<!T>::'<NonNullProperty>k__BackingField'
IL_0006:  dup
IL_0007:  brtrue.s   IL_0015
IL_0009:  pop
IL_000a:  ldstr      "[NullGuard] Return value of property 'T GenericCla"
+ "ssWithReferenceTypeConstraints`1::NonNullProperty()' is null."
IL_000f:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
IL_0014:  throw
IL_0015:  ret
}
.method public hidebysig specialname instance void
set_NonNullProperty(!T 'value') cil managed
{
.maxstack  2
IL_0000:  ldarg.1
IL_0001:  brtrue.s   IL_0013
IL_0003:  ldstr      "value"
IL_0008:  ldstr      "[NullGuard] Cannot set the value of property 'T Ge"
+ "nericClassWithReferenceTypeConstraints`1::NonNullProperty()' to null."
IL_000d:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
string)
IL_0012:  throw
IL_0013:  ldarg.0
IL_0014:  ldarg.1
IL_0015:  stfld      !0 class GenericClassWithReferenceTypeConstraints`1<!T>::'<NonNullProperty>k__BackingField'
IL_001a:  ret
}
.method public hidebysig specialname instance !T
get_CanBeNullProperty() cil managed
{
.maxstack  8
IL_0000:  ldarg.0
IL_0001:  ldfld      !0 class GenericClassWithReferenceTypeConstraints`1<!T>::'<CanBeNullProperty>k__BackingField'
IL_0006:  ret
}
.method public hidebysig specialname instance void
set_CanBeNullProperty(!T 'value') cil managed
{
.maxstack  8
IL_0000:  ldarg.0
IL_0001:  ldarg.1
IL_0002:  stfld      !0 class GenericClassWithReferenceTypeConstraints`1<!T>::'<CanBeNullProperty>k__BackingField'
IL_0007:  ret
}
.method public hidebysig instance !T  NonNullMethod() cil managed
{
.maxstack  2
IL_0000:  ldarg.0
IL_0001:  call       instance !0 class GenericClassWithReferenceTypeConstraints`1<!T>::get_CanBeNullProperty()
IL_0006:  dup
IL_0007:  brtrue.s   IL_0015
IL_0009:  pop
IL_000a:  ldstr      "[NullGuard] Return value of method 'T GenericClass"
+ "WithReferenceTypeConstraints`1::NonNullMethod()' is null."
IL_000f:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
IL_0014:  throw
IL_0015:  ret
}
.method public hidebysig instance !!U  GenericMethod<([mscorlib]System.Collections.IList) U>(!T t,
!!U u) cil managed
{
.maxstack  2
.locals init (!!U V_0)
IL_0000:  ldarg.1
IL_0001:  brtrue.s   IL_0013
IL_0003:  ldstr      "t"
IL_0008:  ldstr      "[NullGuard] t is null."
IL_000d:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
string)
IL_0012:  throw
IL_0013:  ldarg.2
IL_0014:  brtrue.s   IL_0026
IL_0016:  ldstr      "u"
IL_001b:  ldstr      "[NullGuard] u is null."
IL_0020:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
string)
IL_0025:  throw
IL_0026:  ldloca.s   V_0
IL_0028:  initobj    !!U
IL_002e:  ldloc.0
IL_002f:  dup
IL_0030:  brtrue.s   IL_003e
IL_0032:  pop
IL_0033:  ldstr      "[NullGuard] Return value of method 'U GenericClass"
+ "WithReferenceTypeConstraints`1::GenericMethod(T,U)' is null."
IL_0038:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
IL_003d:  throw
IL_003e:  ret
}
.method public hidebysig instance !!U  GenericMethodReturnsParameter<([mscorlib]System.Collections.IList) U>(!T t,
!!U u) cil managed
{
.maxstack  2
IL_0000:  ldarg.1
IL_0001:  brtrue.s   IL_0013
IL_0003:  ldstr      "t"
IL_0008:  ldstr      "[NullGuard] t is null."
IL_000d:  newobj     instance void [mscorlib]System.ArgumentNullException::.ctor(string,
string)
IL_0012:  throw
IL_0013:  ldarg.2
IL_0014:  dup
IL_0015:  brtrue.s   IL_0023
IL_0017:  pop
IL_0018:  ldstr      "[NullGuard] Return value of method 'U GenericClass"
+ "WithReferenceTypeConstraints`1::GenericMethodReturnsParameter(T,U)' is "
+ "null."
IL_001d:  newobj     instance void [mscorlib]System.InvalidOperationException::.ctor(string)
IL_0022:  throw
IL_0023:  ret
}
.method public hidebysig specialname rtspecialname
instance void  .ctor() cil managed
{
.maxstack  1
IL_0000:  ldarg.0
IL_0001:  call       instance void [mscorlib]System.Object::.ctor()
IL_0006:  ret
}
.property instance !T NonNullProperty()
{
.get instance !T GenericClassWithReferenceTypeConstraints`1::get_NonNullProperty()
.set instance void GenericClassWithReferenceTypeConstraints`1::set_NonNullProperty(!T)
}
.property instance !T CanBeNullProperty()
{
.custom instance void CanBeNullAttribute::.ctor() = ( 01 00 00 00 )
.get instance !T GenericClassWithReferenceTypeConstraints`1::get_CanBeNullProperty()
.set instance void GenericClassWithReferenceTypeConstraints`1::set_CanBeNullProperty(!T)
}
}
